src/library/scala/math/ScalaNumericConversions.scala

```diff
@@ -13,58 +13,58 @@ package math
  *  extend ScalaNumber (which excludes value classes.)
  */
 trait ScalaNumericConversions extends ScalaNumber with ScalaNumericAnyConversions {
-  def underlying(): Object
+  def underlying: Object
 }
 
 /** Conversions which present a consistent conversion interface
  *  across all the numeric types, suitable for use in value classes.
  */
 trait ScalaNumericAnyConversions extends Any {
   /** @return `'''true'''` if this number has no decimal component, `'''false'''` otherwise. */
-  def isWhole(): Boolean
-  def underlying(): Any
+  def isWhole: Boolean
+  def underlying: Any
 
-  def byteValue(): Byte
-  def shortValue(): Short
-  def intValue(): Int
-  def longValue(): Long
-  def floatValue(): Float
-  def doubleValue(): Double
+  def byteValue: Byte
+  def shortValue: Short
+  def intValue: Int
+  def longValue: Long
+  def floatValue: Float
+  def doubleValue: Double
```

Comment:
Those mimic the methods of `java.lang.Number`, which, being Java methods, have `()`. IMO they should have `()` as well.

Reply:
For clarity, what's the drawback on them not having `()`? That it's an extra migration step when porting Java code to Scala?

Reply:
But Dotty lets you call methods that are defined in Java or that override methods defined in Java with or without `()`. The methods above, you would not be able to call them without `()`. So in a sense, it is inconsistent with the methods from `java.lang.Number`.

Reply:
The way you defined them, I cannot call them *with* `()`, whereas I can do so with `java.lang.Number`. It's inconsistent either way. This whole maybe-`()` thing is a huge mess of inconsistency. The worst is people *overriding* Java methods that have `()` with methods that don't (e.g., `override def toString: String`) because you cannot call them as `x.toString()` anymore, whereas you can on their super types -> violation of LSP.

The only thing that would really save us at this point would be:
* enforce the proper `()` when overriding, including when overriding Java methods
* at call site, demand `()` if they are at definition site, *except* if the method is Java-defined *or* it overrides a Java-defined method

In the meantime, we should at least strive for consistency in the way we *defined* our methods, even if call sites are a huge mess.

Reply:
> The worst is people overriding Java methods that have () with methods that don't (e.g., override def toString: String) because you cannot call them as x.toString() anymore, whereas you can on their super types -> violation of LSP.

You can call call them as `x.toString()` since they override a Java defined method

Reply:
Uh ... looks like the mess is even bigger than I thought. But OK, whatever. I've always thought I didn't have the energy for this battle.

Reply:
> Uh ... looks like the mess is even bigger than I thought

Isn't Dotty's rule at call site the rule you suggested above?
> at call site, demand () if they are at definition site, except if the method is Java-defined or it overrides a Java-defined method

Reply:
Regarding your first point, we already do it for Scala methods
> enforce the proper () when overriding, including when overriding Java methods

Reply:
Yeah it's my first point which should also apply to Java methods, IMO. But I can see how dotty's situation is rational and defensible.

Reply:
Currently, if my `def f` overrides `def f()`, my `f` gets an empty param list. For example, the added parens to the definition of `ReferenceWrapper.enqueue()` were already implicit. Will that still be true in future? That's 5.1.4 of SLS, "A special rule...".